packages/zoe/src/z2spec/escrow-exo.js

```diff
@@ -0,0 +1,311 @@
+      seat: {
+        /** @param {Keyword} kw */
+        async getPayout(kw) {
+          const { payoutsKit } = this.state;
+          const paid = await payoutsKit.subscriber.getUpdateSince();
+          return paid.value[kw];
+        },
+        async getPayouts() {
+          const { payoutsKit } = this.state;
+          const paid = await payoutsKit.subscriber.getUpdateSince();
+          return paid.value;
+        },
+
+        /** @param {unknown} completion */
+        async exit(completion) {
+          assert.fail('TODO');
+        },
+        fail(reason) {
+          assert.fail('TODO');
+        },
+
+        readOnly() {
+          const { readOnly } = this.facets;
+          return readOnly;
+        },
+
+        getInvitationCustomDetails() {
+          const { exiting, proposal, currentAllocation } = this.state;
+          !exiting || Fail`already exited`;
+
+          return harden({
+            proposal,
+            currentAllocation,
+          });
+        },
+
+        // repeat accessors for compat
+        hasExited() {
+          return this.state.exiting;
+        },
+        getProposal() {
+          return this.state.proposal;
+        },
+        getCurrentAllocation() {
+          return this.state.currentAllocation;
+        },
+        getExitSubscriber() {
+          return this.state.exitKit.subscriber;
+        },
+      },
+      /** The right to allocate payouts (subject to offer safety) */
+      resolver: {
+        readOnly() {
+          const { readOnly } = this.facets;
+          return readOnly;
+        },
+        /**
+         * @param {AmountKeywordRecord} more
+         * @param {AmountKeywordRecord} less
+         * @throws if proposal is violated
+         */
+        updateAllocation(more, less) {
+          const { proposal, currentAllocation, exiting } = this.state;
+          !exiting || Fail`already exited`;
+          const candidate = subtractFromAllocation(
+            addToAllocation(currentAllocation, more),
+            less,
+          );
+          isOfferSafe(proposal, candidate) ||
+            Fail`Offer safety was violated by the proposed allocation: ${candidate}. Proposal was ${proposal}`;
+          this.state.currentAllocation = candidate;
+        },
+        /** @param {unknown} completion */
+        async exit(completion) {
+          if (this.state.exiting) {
+            return;
+          }
+          const { exitKit, payoutsKit } = this.state;
+
+          this.state.exiting = true;
+          exitKit.publisher.finish(completion);
+          await deeplyFulfilledObject(payoutsKit.subscriber.getUpdateSince());
+        },
+        fail(reason) {
+          assert.fail('TODO');
+        },
+
+        getInvitationCustomDetails() {
+          const { exiting, proposal, currentAllocation } = this.state;
+          !exiting || Fail`already exited`;
+
+          return harden({
+            proposal,
+            currentAllocation,
+          });
```

Comment:
How does one distinguish an invitation for an ownable seat vs an ownable seat resolver?